frontend/src/container/PublicProfile.jsx

```diff
@@ -73,32 +102,178 @@ export class PublicProfile extends React.Component {
+    const uiSchemaObjectRequest = this.getUserCustomPropertiesUiSchema()
+    const dataSchemaObjectRequest = this.getUserCustomPropertiesDataSchema(userId)
+
+    const [schemaObject, uiSchemaObject, dataSchemaObject] = await Promise.all(
+      [schemaObjectRequest, uiSchemaObjectRequest, dataSchemaObjectRequest]
+    )
+
+    const [informationSchema, personalPageSchema] = this.splitSchema(schemaObject, uiSchemaObject)
+    const [informationDataSchema, personalPageDataSchema] = this.splitDataSchema(dataSchemaObject, uiSchemaObject)
+
+    this.setState({
+      informationSchemaObject: informationSchema,
+      personalPageSchemaObject: personalPageSchema,
+      uiSchemaObject: uiSchemaObject,
+      informationDataSchema: informationDataSchema,
+      personalPageDataSchema: personalPageDataSchema,
+      dataSchemaObject: dataSchemaObject
+    })
+  }
+
+  splitSchema = (schema, uiSchema) => {
+    const informationSchema = {
+      ...schema,
+      required: schema.required.filter(field =>
+        this.findPropertyDisplayGroup(field, uiSchema) === DISPLAY_GROUP_BACKEND_KEY.information
+      ),
+      properties: Object.fromEntries(
+        Object.entries(schema.properties).filter(([key, val]) =>
+          this.findPropertyDisplayGroup(key, uiSchema) === DISPLAY_GROUP_BACKEND_KEY.information
+        )
+      )
+    }
+    const personalPageSchema = {
+      ...schema,
+      title: '', // INFO - CH - 20210122 - reset title and description since they are used for first form
+      description: '',
+      required: schema.required.filter(field =>
+        this.findPropertyDisplayGroup(field, uiSchema) === DISPLAY_GROUP_BACKEND_KEY.personalPage
+      ),
+      properties: Object.fromEntries(
+        Object.entries(schema.properties).filter(([key, val]) =>
+          this.findPropertyDisplayGroup(key, uiSchema) === DISPLAY_GROUP_BACKEND_KEY.personalPage
+        )
+      )
+    }
+    return [informationSchema, personalPageSchema]
+  }
+
+  splitDataSchema = (dataSchema, uiSchema) => {
+    const informationDataSchema = {
+      ...Object.fromEntries(
+        Object.entries(dataSchema).filter(([key, value]) =>
+          this.findPropertyDisplayGroup(key, uiSchema) === DISPLAY_GROUP_BACKEND_KEY.information
+        )
+      )
+    }
+    const personalPageDataSchema = {
+      ...Object.fromEntries(
+        Object.entries(dataSchema).filter(([key, value]) =>
+          this.findPropertyDisplayGroup(key, uiSchema) === DISPLAY_GROUP_BACKEND_KEY.personalPage
+        )
+      )
+    }
+
+    return [informationDataSchema, personalPageDataSchema]
+  }
+
+  findPropertyDisplayGroup = (property, uiSchema) => {
+    if (!property || !uiSchema || !uiSchema[property]) return ''
+    return uiSchema[property][DISPLAY_GROUP_BACKEND_KEY.uiSchemaKey]
+  }
+
+  getUserCustomPropertiesSchema = async () => {
+    const { props } = this
+    const result = await props.dispatch(getCustomPropertiesSchema())
+    switch (result.status) {
```

Comment:
Are the errors for these requests handled?
If yes feel free to resolve, otherwise we can open an issue

Reply:
I will add this to the issue about things that's left to be done for custom properties

Reply:
issue created https://github.com/tracim/tracim/issues/4089